packages/client/src/net/peer/peer.ts

```diff
@@ -38,17 +33,22 @@ export interface PeerOptions {
  * Network peer
  * @memberof module:net/peer
  */
-export class Peer extends EventEmitter {
+export abstract class Peer extends EventEmitter {
   public config: Config
   public id: string
   public address: string
   public inbound: boolean
   public server: Server | undefined
-  public bound: Map<string, BoundProtocol>
   protected transport: string
   protected protocols: Protocol[]
+  protected boundProtocols: BoundProtocol[] = []
```

Comment:
Is this still an intermediate step to keep this generic `boundProtocols` structure? Otherwise we loose many of the typing benefits here again? 🤔

I thought we can call on `peer.eth.....` and so forth directly?

Reply:
We can call `peer.eth` directly. The reason why I moved this `boundProtocols` here is because it is used to keep track of what protocols we have added. We can actually remove this, and just do simple checks: `if (this.eth !== undefined)` then we know that we have support for eth protocol.

(Not entirely sure if this answers your question / point?)

The only places where it is used is in `toString()` and into `handleMessageQueue`, this `boundProtocols` can easily be removed (and maybe should do so already?).

Reply:
If it's only used on these few occasions that it is no big deal for sure. You are a lot deeper into it, then just decide yourself if you'd rather want to keep or not.

Reply:
I'll open a cleanup PR and will also deal with the Peer/RlpxPeer stuff